Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/MoodHistoryFragment.java

```diff
@@ -96,5 +102,66 @@ public void onMoodHistoryUpdate(ArrayList<MoodEvent> newHistory) {
         moodEventList.addAll(newHistory);
         recyclerAdapter.notifyDataSetChanged();
     }
+
+    /**
+     * Utility class to add swipe support to recycler view for deleting a mood event
+     */
+    ItemTouchHelper.SimpleCallback itemTouchHelperCallback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT) {
+
+        /**
+         * Called when ItemTouchHelper wants to move the dragged item from its old position to the new position
+         * @param recyclerView The RecyclerView to which ItemTouchHelper is attached to
+         * @param viewHolder The ViewHolder which is being dragged by the user
+         * @param target  The ViewHolder over which the currently active item is being dragged
+         * @return If this method returns true, ItemTouchHelper assumes viewHolder has been moved to the adapter position of target ViewHolder, else, it hasn't been moved
+         */
+        @Override
+        public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
+            return false;
+        }
+
+        /**
+         * Called when a ViewHolder is swiped by the user
+         * @param viewHolder The ViewHolder which has been swiped by the user
+         * @param direction The direction to which the ViewHolder is swiped
+         */
+        @Override
+        public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
+            MoodEvent deletedMood = moodEventList.remove(viewHolder.getAdapterPosition());
```

Comment:
Because the mood history list is already listening for changes to Firestore, you don't need this line and "recyclerAdapter.notifyDataSetChanged". The changes will automatically be reflected!